### PR TITLE
Issue #SB-21199 feat: added login functionality for desktop

### PR DESCRIPTION
**Pull request by Nik720**

### CODE REVIEW

File: src/desktop/modules/routes/auth.ts [hunk 1]
```diff
@@ -11,5 +11,8 @@ export default (app, proxyURL) => {
         "/api/auth/resolvePasswordSession/:code", authController.resolvePasswordSession.bind(authController),
     );
 
+    app.post(
```
Comment (HarishGangula) [inline src/desktop/modules/routes/auth.ts:14]: Add version number to api 
It can be **/api/user/v1/startSession**

File: src/desktop/main.ts [hunk 2]
```diff
@@ -205,39 +206,19 @@ expressApp.use("/dialog/content/suggestLocation", async (req, res) => {
...
-  res.send({ message: "SUCCESS", responseCode: "OK" })
+  const options = {"request":{"type":"desktopConfig","action":"get","subType":"login"}};
+  HTTPService.post(`${appBaseUrl}/api/data/v1/form/read`, options).toPromise().then(async response => {
+    let loginFormConfig = _.find(response.data.result.form.data.fields, (field) => field.context == 'login');
+    const loginSessionProvider = new LoginSessionProvider(win, loginFormConfig, appBaseUrl);
+    loginSessionProvider.childLoginWindow().then((rs) =>{
+      res.send({ message: "SUCCESS", responseCode: "OK" })
+    })
+  }, err => {
+    res.send({ message: "Login failed", responseCode: "OK" })
```
Comment (HarishGangula) [inline src/desktop/main.ts:217]: in err case responseCode should not be **OK**

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/desktop/modules/routes/auth.ts (hunk 1 above)
Comment (Nik720) [reply src/desktop/modules/routes/auth.ts:14]: done

File: src/desktop/main.ts (hunk 2 above)
Comment (Nik720) [reply src/desktop/main.ts]: done

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**